scripts/locust/service.py

```diff
@@ -1,30 +1,71 @@
 from contextlib import (
     contextmanager,
 )
+from dataclasses import (
+    dataclass,
+)
+from functools import (
+    lru_cache,
+)
+import json
+import logging
+import os
+import random
 from urllib.parse import (
     urlsplit,
     urlunsplit,
 )
 
+from furl import (
+    furl,
+)
 from gevent.pool import (
     Group,
 )
 from locust import (
-    HttpLocust,
-    TaskSequence,
-    TaskSet,
-    seq_task,
+    HttpUser,
+    SequentialTaskSet,
+    between,
     task,
 )
+from more_itertools import (
+    one,
+)
 
+from azul import (
+    config,
+    require,
+)
+from azul.logging import (
+    configure_script_logging,
+)
+from azul.types import (
+    JSON,
+)
 
-# To run:
-#  - make sure locust is installed
-#  - `locust -f scripts/locust/service.py`
-#  - in browser go to localhost:8089
+# This script simulates a user triggering Azul endpoints via the data browser
+# GUI.
+
+# Usage:
+#  - Set $AZUL_LOCUST_CATALOG to the desired catalog, or leave unset or blank to
+#    test the default catalog. Note that the DRS/DOS endpoint tests always use
+#    the DCP1 catalog regardless of this setting.
+#  - Run `locust -f scripts/locust/service.py`
+#  - In browser go to localhost:8089
 #
 # For more info see https://docs.locust.io/en/stable/
 
+log = logging.getLogger(__name__)
+configure_script_logging(log)
+
+
+@lru_cache
+def configured_catalog() -> str:
+    # Locust does not support passing command-line arguments to the script
+    catalog = os.environ.get('AZUL_LOCUST_CATALOG') or config.default_catalog
```

Comment:
Ambiguous. Empty string should be rejected.

Reply:
Also, is there really no other way to parameterize the task set? Can it be done through the web UI?

Reply:
I can't find any other way. There is a [github thread on this topic](https://github.com/locustio/locust/issues/65), which was where I got the idea for this approach in the first place. It seems our options are:

1. Use the command-line var
2. Read config from a file such as (`environment.local.py`?)
3. Fork locust or use someone's existing fork

Reply:
We'll stick with env var in that case. Please make sure that the environment variables doesn't get exported to the lambda. I think the convention is to lower case name but you should double check. It may also be leading underscore, I don't recall.

Reply:
Switched to lower case. From `environment.py`: 
> Only variables whose names start in `AZUL_` will be published to a deployed Lambda. Note that by implication, `azul_` variables will not be published, even though they are considered part of Azul.